### PR TITLE
Rate limit CPU warning logs from the Voice Focus worklet

### PR DESCRIPTION
**Pull request by richhx**
**Issue #:**

#2241

**Description of changes:**

Add basic rate limiting to log CPU warnings at most every minute. This is to prevent dumping logs to the console. Technically, we already have an implementation that will stop the Voice Focus worklet processor in the case that it gets too far behind, but in some cases, it's not fast enough. In general, builders can always use the CPU delegate in case they want to add more logs or add additional business logic.

**Testing:**
Verify with a CPU limited browser and the higher complexity, using inline execution, and no SIMD support that the logs still work and are rate limited.

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
N/A

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
Yes.

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
No.

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
No.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,17 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
```
Comment (ltrung) [inline CHANGELOG.md:8]: You no longer need to do that. The current active development version (unreleased in npm) is the latest versionin the CHANGELOG.

File: libs/voicefocus/voicefocus.js [hunk 2]
```diff
@@ -23,12 +26,12 @@ const DEFAULT_AGC_DISABLED_SETTING = {
...
 const DEFAULT_AGC_SETTING = DEFAULT_AGC_DISABLED_SETTING;
 const DEFAULT_ASSET_GROUP = 'stable-v1';
-const DEFAULT_CDN = 'https://static.sdkassets.chime.aws/';
+const DEFAULT_CDN = 'https://static.sdkassets.chime.aws';
 const DEFAULT_PATHS = {
-    processors: `${DEFAULT_CDN}processors/`,
-    workers: `${DEFAULT_CDN}workers/`,
-    wasm: `${DEFAULT_CDN}wasm/`,
-    models: `${DEFAULT_CDN}wasm/`,
+    processors: `${DEFAULT_CDN}/processors/`,
```
Comment (ltrung) [inline libs/voicefocus/voicefocus.js:31]: just curious, what is the benefit of this change?

File: libs/voicefocus/worklet-inline-node.js [hunk 3]
```diff
@@ -59,7 +64,15 @@ class VoiceFocusInlineNode extends types_js_1.VoiceFocusAudioWorkletNode {
         const data = event.data;
         switch (data.message) {
             case 'cpu':
```
Comment (ltrung) [inline libs/voicefocus/worklet-inline-node.js:66]: Not familiar with this code so where do we get these cpu message from?

File: libs/voicefocus/worklet-inline-node.js [hunk 4]
```diff
@@ -11,11 +14,13 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
 Object.defineProperty(exports, "__esModule", { value: true });
 const support_js_1 = require("./support.js");
 const types_js_1 = require("./types.js");
+const CPU_WARNING_MAX_INTERVAL_MILLISECONDS = 60 * 1000;
```
Comment (ltrung) [inline libs/voicefocus/worklet-inline-node.js:17]: nit: We usually just use MS instead

Comment (ltrung) [reply libs/voicefocus/worklet-inline-node.js:17]: Is 60s a bit too long?

File: libs/voicefocus/voicefocus.js (hunk 2 above)
Comment (richhx) [reply libs/voicefocus/voicefocus.js:31]: No benefit except I think it's cleaner / easier to read.

File: libs/voicefocus/worklet-inline-node.js (hunk 4 above)
Comment (richhx) [reply libs/voicefocus/worklet-inline-node.js:17]: I think 60 is ok, but I can change it to 1 or 10 to make it more apparent to builders in logs that voice focus is not working well due to CPU usage.

File: CHANGELOG.md (hunk 1 above)
Comment (ltrung) [reply CHANGELOG.md]: You should add your change to 3.4.0 (next version) once you rebase.

File: libs/voicefocus/worklet-inline-node.d.ts [hunk 5]
```diff
@@ -3,6 +3,8 @@ declare class VoiceFocusInlineNode extends VoiceFocusAudioWorkletNode {
     private delegate?;
     private worker;
     private logger;
+    private cpuWarningLastTriggered;
+    private cpuWarningCount;
```
Comment (devalevenkatesh) [inline libs/voicefocus/worklet-inline-node.d.ts:7]: Can we initialize this value  to 0 here itself?

File: CHANGELOG.md [hunk 6]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Replace the deprecated API `getRTCPeerConnectionStats` with `metricsDidReceive` in meeting readiness checker. 
 - Prevent `realtimeUnsubscribeFromVolumeIndicator` from causing a fatal error when there are no subscriptions for the `attendeeId`.
 - Subscribe to `audioOutputDidChange` in audio mix controller to fix the issue where the audio output is not updated before meeting start.
+- Rate limited CPU warnings to at most once a minute in Voice Focus library, so that builder logs are not flooded.
```
Comment (ltrung) [inline CHANGELOG.md:21]: can this move to 3.4.0 section?

File: libs/voicefocus/worklet-inline-node.js [hunk 7]
```diff
@@ -1,3 +1,6 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (richhx) [reply libs/voicefocus/worklet-inline-node.js:1]: Yes it is generated. We have an internal package that does the transpiling and it's copied here.

File: libs/voicefocus/support.js [hunk 8]
```diff
@@ -132,3 +135,9 @@ const isOldChrome = (global = globalThis, logger) => {
     return false;
 };
 exports.isOldChrome = isOldChrome;
+const getDateDiffInMilliseconds = (a, b) => {
```
[the remark this answers is not in the document]
Comment (richhx) [reply libs/voicefocus/support.js:138]: Good point. Switched it to use epoch ms time instead.

File: CHANGELOG.md (hunk 6 above)
Comment (richhx) [reply CHANGELOG.md:21]: Yep will rebase.

File: CHANGELOG.md [hunk 9]
```diff
@@ -30,6 +31,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
```
Comment (devalevenkatesh) [inline CHANGELOG.md:34]: Could you please remove this one since you added under 3.4.0?

**Review by ltrung — APPROVED** [review]

**Review by devalevenkatesh — APPROVED** [review]

**Pull request merged**